### PR TITLE
Dependency updates

### PR DESCRIPTION
**Pull request by mshima**
Fixes https://github.com/jhipster/generator-jhipster-react-native/pull/1229.

- drop npmrc file. It's used for legacy-peer-deps.
Peer dependencies should be supported by every non deprecated lib nowadays.
- don't install npm 7 which is outdated nowadays, node 16 bundles npm 8.
- replace deprecated enzyme with testing-library.
- remove old package-lock and node_modules on update, npm is not good resolving conflicts with a completely new package.json dependencies.

### CODE REVIEW

**Comment by mraible** [comment]
@mshima It looks like your last commit had all tests passing. After committing https://github.com/jhipster/generator-jhipster-react-native/pull/1254/commits/6124dcc9cd43c8003b3164881085c60a0569ac3f, tests are failing. Are there some dependencies that shouldn't be updated?

**Comment by mshima** [comment]
The failure is due to a transitive dependency.
At my sample recreating node modules breaks tests.
I couldn’t find it yet. My guess is jest or babel.

File: generators/app/templates/package.json [hunk 1]
```diff
@@ -37,18 +38,18 @@
...
     "create-expo-app": "2.0.3",
     "detox": "20.11.2",
-    "enzyme": "3.11.0",
-    "enzyme-adapter-react-16": "1.15.7",
     "eslint": "8.46.0",
     "eslint-config-prettier": "8.9.0",
     "eslint-plugin-prettier": "4.2.1",
     "expect": "29.6.2",
-    "expo": "47.0.13",
+    "expo": "48.0.17",
```
Comment (mshima) [inline generators/app/templates/package.json:52]:
```suggestion
    "expo": "49.0.6",
```

File: generators/app/templates/package.expo.json [hunk 2]
```diff
@@ -4,25 +4,25 @@
...
+    "react-native-reanimated": "~2.14.4",
+    "react-native-safe-area-context": "4.5.0",
+    "react-native-screens": "~3.20.0",
     "react-native-web": "^0.18.9"
   },
   "devDependencies": {
-    "@expo/webpack-config": "^0.17.2",
-    "jest-expo": "^46.0.1"
+    "@expo/webpack-config": "^18.0.1",
+    "jest-expo": "^48.0.2"
```
Comment (mshima) [inline generators/app/templates/package.expo.json:26]:
```suggestion
    "jest-expo": "^49.0.0"
```

**Comment by mshima** [comment]
Now heroku seem to be timing out..

**Comment by mraible** [comment]
@mshima Can you please resolve conflicts?

**Comment by mshima** [comment]
@mraible is dependabot merging PRs with failures?
Shouldn't tests be required to pass?

**Comment by mraible** [comment]
@mshima Good eye! Branch protection wasn't enabled. I enabled it and checked all the CI jobs. I'll also add you as an admin to this repo.

Screenshot 2023-08-16 at 9 12 03 PM

**Pull request merged**